### PR TITLE
Statements: return dumping methods.

### PR DESCRIPTION
**Pull request by sandreenko**
Part of #25899 

5e4b1248d5: Return dumping for Statement.

Return statements ID and implement dumping methods.

a0bda39c05: Delete `block->bbStmtNum` and `compiler->compCurStmtNum`.

They were duplicating `stmt->gtTreeID` in the past and it was not clear why they were added in the first place.

Delete them and use `stmt->GetID()`, with `STMT%05u` format.

21ef8c78fc: Use `FMT_STMT` for printing `Statement->GetID()`.

So you can use `STMT\d{5}` to seach them anywhere in JiDump.

### CODE REVIEW

**Comment by sandreenko** [comment]
PTAL @dotnet/jit-contrib

File: src/jit/flowgraph.cpp [hunk 1]
```diff
@@ -20520,23 +20520,17 @@ void Compiler::fgDispBasicBlocks(bool dumpTrees)
...
 }
 
-/*****************************************************************************/
-//  Increment the stmtNum and dump the tree using gtDispTree
+//------------------------------------------------------------------------
+// fgDumpStmtTree: dump the statement and the basic block number.
+//
+// Arguments:
+//    stmt  - the statement to dump;
+//    bbNum - The basic block number to dump.
```
Comment (erozenfeld) [inline src/jit/flowgraph.cpp]: Nit: capitalization of the comments is inconsistent (`the` vs. 'The`).

File: src/jit/gentree.cpp [hunk 2]
```diff
@@ -11468,6 +11468,37 @@ void Compiler::gtDispArgList(GenTreeCall* call, IndentStack* indentStack)
     }
 }
 
+void Compiler::gtDispStmt(Statement* stmt, const char* msg /* = nullptr */)
```
Comment (erozenfeld) [inline src/jit/gentree.cpp]: Please add a header.

File: src/jit/gentree.h [hunk 3]
```diff
@@ -5129,17 +5129,26 @@ struct GenTreeILOffset : public GenTree
 #endif
 };
 
+// We use the following format when print the Statement number: Statement->GetID()
```
Comment (erozenfeld) [inline src/jit/gentree.h]: print --> printing

**Comment by sandreenko** [comment]
Thanks, @erozenfeld , the comments were fixed.

**Pull request merged**